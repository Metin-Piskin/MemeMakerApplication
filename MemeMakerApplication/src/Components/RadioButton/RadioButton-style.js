import { StyleSheet } from "react-native";


export default StyleSheet.create({
    radiocontainer: {
        flexDirection: 'row',
        justifyContent: 'flex-end',
        paddingHorizontal: 10
    },
    text: {
        alignSelf: 'center',
        color: '#fff'
    }
})