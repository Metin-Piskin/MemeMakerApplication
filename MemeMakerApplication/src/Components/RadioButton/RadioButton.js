import React from 'react';
import { View, Text } from 'react-native';
import { RadioButton } from 'react-native-paper';

import styles from './RadioButton-style';

const RadioButtonComponetn = ({ text, value, onPress, status }) => {
    return (
        <View style={styles.radiocontainer} testID='radio-button-container'>
            <Text style={styles.text} testID='radio-button-text'>
                {text}
            </Text>
            <RadioButton
                value={value}
                onPress={onPress}
                status={status}
                color='#fff'
                uncheckedColor='#fff'
            />
        </View>
    )
}

export default RadioButtonComponetn;