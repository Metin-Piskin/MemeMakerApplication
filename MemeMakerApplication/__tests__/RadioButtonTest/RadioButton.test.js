import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import RadioButton from '../../src/Components/RadioButton';
import RadioButtonStyles from '../../src/Components/RadioButton/RadioButton-style';

jest.mock('react-native-paper')

test('renders correctly', () => {
    const comp = render(<RadioButton />);
    expect(comp).toMatchSnapshot();
});

test('should render title correctly', () => {
    const test = 'test';
    const comp = render(<RadioButton text={test} />);

    const Text = comp.getByTestId('radio-button-text').children[0];
    expect(Text).toBe(test);
});

test('should render default theme style', () => {
    const comp = render(<RadioButton />);
    const RadioButtonContainerStyle = comp.getByTestId('radio-button-container').props.style;

    expect(RadioButtonContainerStyle).toMatchObject(RadioButtonStyles.radiocontainer);
});

test('should render default theme style', () => {
    const comp = render(<RadioButton />);
    const RadioButtonTextStyle = comp.getByTestId('radio-button-text').props.style;

    expect(RadioButtonTextStyle).toMatchObject(RadioButtonStyles.text);
});
