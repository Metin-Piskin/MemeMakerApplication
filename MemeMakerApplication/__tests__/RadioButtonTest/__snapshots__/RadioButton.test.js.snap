// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "paddingHorizontal": 10,
    }
  }
  testID="radio-button-container"
>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#fff",
      }
    }
    testID="radio-button-text"
  />
</View>
`;
